generators/server/templates/src/main/java/package/config/jcache/_JCacheGaugeSet.java

```diff
@@ -0,0 +1,76 @@
+package <%=packageName%>.config.jcache;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.JmxAttributeGauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+
+import java.lang.management.ManagementFactory;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.cache.management.CacheStatisticsMXBean;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectInstance;
+import javax.management.ObjectName;
+
+import static com.codahale.metrics.MetricRegistry.*;
+
+/**
+ * MetricSet retriving JCache specific JMX metrics for every configured caches
+ */
+public class JCacheGaugeSet implements MetricSet {
+
+    private final Logger log = LoggerFactory.getLogger(JCacheGaugeSet.class);
+
+    private static final String M_BEAN_COORDINATES = "javax.cache:type=CacheStatistics,CacheManager=*,Cache=*";
+
+    private Set<ObjectInstance> objectInstances;
+
+    public JCacheGaugeSet() {
+        try {
+            this.objectInstances = ManagementFactory.getPlatformMBeanServer().queryMBeans(ObjectName.getInstance(M_BEAN_COORDINATES), null);
+        } catch (MalformedObjectNameException e) {
+            log.error("Failed to retrieve JCache statistics", e);
```

Comment:
The exception shouldn't happen since I'm providing the query. But I did a bunch of refactoring to make the code simpler and I am now throwing an `InternalError`